### ISSUE TITLE
The page numbers don't seem to be updating from 0 in new or edit mode. It is unclear to me why at the moment. I think it may have something to do with the number formatter but not sure.

### ISSUE THREAD

**Issue by LincolnDickerson** [peterfriese/BookSpine#3]
Additionally the number field when edited does not toggle the save button to on.

**Comment by peterfriese** [comment]
You're absolutely right - it's due to the way how the `NumberFormatter` works. I've mentioned this in my [article](https://peterfriese.dev/swiftui-firebase-add-data/):

> Please note that, as of now, you need to press [Enter] to commit any input in a TextField that uses a Formatter subclass. This likely is Apple’s attempt to minimize the number of times the formatter needs to run.

And in the [video as well](https://youtu.be/imTjGvSjBEw?t=658)

There is not much we can do about it now, unfortunately.

**Comment by LincolnDickerson** [comment]
Thanks for the quick reply. 

Truth is I used this example to guide me through my last project that used firebase and SwiftUI. I am currently starting a new one and wanted to review the the example so I just cloned it and ran with it. 

I think Apple dropped the ball here. I will like use preferences instead of the formatter option because of this shortcoming. 

Thank you very much for all your contribution to the community.  I have learned from many of your videos. 

Have a great day and happy coding!
